rdr_service/resource/generators/participant.py

```diff
@@ -167,6 +171,35 @@ def _prep_participant(self, p_id, ro_session):
 
         return data
 
+    def _prep_participant_profile(self, p_id, ro_session):
+        """
+        Get additional participant status fields that were incorporated into the RDR participant_summary
+        but can't be derived from other RDR tables.  Example is EHR status information which is
+        read from a curation dataset by a daily cron job that then applies updates to RDR participant_summary directly.
+        :param p_id: participant_id
+        :return: dict
+
+        """
+        # TODO: Workaround for PDR-106 is to pull needed EHR fields from participant_summary. LIMITED USE CASE ONLY
+        # Goal is to eliminate dependencies on participant_summary, which may go away someday.
+        # Long term solution may mean creating a participant_profile table for these outlier fields that are managed
+        # outside of the RDR API, and query that table instead.
+        data = {}
+        ps = ro_session.query(ParticipantSummary.ehrStatus, ParticipantSummary.ehrReceiptTime,
+                              ParticipantSummary.ehrUpdateTime) \
+            .filter(ParticipantSummary.participantId == p_id).first()
+
+        if ps and ps.ehrStatus:
+            ehr_status = EhrStatus(ps.ehrStatus)
+            data = {
+                'ehr_status': str(ehr_status),
+                'ehr_status_id': int(ehr_status),
+                'ehr_receipt': ps.ehrReceiptTime,
+                'ehr_update': ps.ehrUpdateTime
+            }
+
+        return data
+
```

Comment:
@robabram, I think we've talked about this before, but I don't remember what you've said about it. Would it make sense to have a common module for the Resource generator and BigQuery DAO to share code like this, rather than having duplication between them? If there's field naming difference them then maybe mapping code would help.

Reply:
The BigQuery DAO stuff was supposed to be gone by now and replaced by the Resource APIs.